Spigot-Server-Patches/0543-Add-PrepareGrindstoneEvent.patch

```diff
@@ -0,0 +1,71 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jan Tuck <jan@tuck.dk>
+Date: Fri, 19 Jun 2020 17:35:24 +0200
+Subject: [PATCH] Add PrepareGrindstoneEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/ContainerGrindstone.java b/src/main/java/net/minecraft/server/ContainerGrindstone.java
+index ed88e208d00f3b19b5c5916cf87513ab511a5fb1..d6ec18358699425ee5f7b7a6c9b2faac68d30902 100644
+--- a/src/main/java/net/minecraft/server/ContainerGrindstone.java
++++ b/src/main/java/net/minecraft/server/ContainerGrindstone.java
+@@ -10,6 +10,10 @@ import org.bukkit.craftbukkit.inventory.CraftInventoryGrindstone;
+ import org.bukkit.craftbukkit.inventory.CraftInventoryView;
+ import org.bukkit.entity.Player;
+ // CraftBukkit end
++// Paper start
++import com.destroystokyo.paper.event.inventory.PrepareGrindstoneEvent;
++import org.bukkit.craftbukkit.inventory.CraftItemStack;
++// Paper end
+ 
+ public class ContainerGrindstone extends Container {
+ 
+@@ -153,12 +157,13 @@ public class ContainerGrindstone extends Container {
+         ItemStack itemstack1 = this.craftInventory.getItem(1);
+         boolean flag = !itemstack.isEmpty() || !itemstack1.isEmpty();
+         boolean flag1 = !itemstack.isEmpty() && !itemstack1.isEmpty();
++        PrepareGrindstoneEvent event = new PrepareGrindstoneEvent(this.getBukkitView(), null); // Paper
+ 
+         if (flag) {
+             boolean flag2 = !itemstack.isEmpty() && itemstack.getItem() != Items.ENCHANTED_BOOK && !itemstack.hasEnchantments() || !itemstack1.isEmpty() && itemstack1.getItem() != Items.ENCHANTED_BOOK && !itemstack1.hasEnchantments();
+ 
+             if (itemstack.getCount() > 1 || itemstack1.getCount() > 1 || !flag1 && flag2) {
+-                this.resultInventory.setItem(0, ItemStack.a);
```

Comment:
There are many places like this where the event result is set but the function returns early. I don't think this is firing the event here due to the early return.

Reply:
I see, that was an oversight on my part